core/textile/bucket/crypto/crypto.go

```diff
@@ -0,0 +1,177 @@
+package crypto
+
+import (
+	"bytes"
+	b64 "encoding/base64"
+	"errors"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/FleekHQ/space-daemon/log"
+)
+
+func parseKeys(key []byte) (aesKey, iv, hmacKey []byte, err error) {
+	if len(key) != aesKeySize+ivKeySize+hmacKeySize {
+		return nil, nil, nil, errors.New("unsupported encryption keys provided.")
+	}
+
+	return key[:aesKeySize], key[aesKeySize:(aesKeySize + ivKeySize)], key[(aesKeySize + ivKeySize):], nil
+}
+
+// EncryptPathItems returns an encrypted path and a Reader that reads the encrypted data from the
+// plain reader passed into the function.
+// Encrypted data is AES-CTR of data + AES-512 HMAC of encrypted data
+//
+// NOTE: key must be a 64 byte long key
+// To decrypt the result of this function use the DecryptPathItems function
+func EncryptPathItems(key []byte, path string, plainReader io.Reader) (string, io.Reader, error) {
+	// split key into key and secret
+	// use key and secret and IV
+
+	// encrypt path
+	log.Debug("Encrypting Path Items", "path:"+path)
+	aesKey, iv, hmacKey, err := parseKeys(key)
+	if err != nil {
+		return "", nil, err
+	}
+
+	encryptedPath := ""
+	pathParts := strings.Split(path, "/")
```

Comment:
Are these paths bucket paths or os paths? Because if it's the latter maybe it should use `path/filepath` package to use cross-platform path delimiters

Reply:
I'm assuming at this level we are dealing with bucket paths, so all transformation should have been done before a request is passed down to the buckets, since non unix `/` paths don't work with bucket I'm not using the filepath or os specific path separator so it doesn't fail for other operating systems